### DISCUSSION TITLE
Big issue with saved routes

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#527] [Q&A] [answered]
all routes are mixed up and partly deleted. I have created several new groups and wanted to put routes into the new groups by using "change group" on a route. There was a message on saving "attempt to read..." repeatedly. I don't know if this is normal or is an error message. Afterwards all routes were mixed up and partly deleted, also in folders which I didn't change. I tried to save my routes again from another device, first in json format, then in kurviger format, but the import is also not working any more. Nothing happens then, no error message, just nothing imported.

By creating new groups I have more than 10 groups. Can it be that there is a problem with the alphanumeric sorting and then groups are mixed up or is it maybe a read/write issue on the file storage?

**Comment by devemux86** [comment]
@DolomitesTom thanks for the report.

What app version?

You can clear data in Android app settings on the device with the problem and start over again.

If you can reproduce it every time, can you describe the detailed steps so we can see what's going on?

**Comment by DolomitesTom** [comment]
Latest version 4.0

When I clear the Android App settings, is then reading all data, routes and favourites in again which it can find on the file storage or do I have to export routes favourites and profiles and import again?

**Reply by devemux86** [reply]
> or do I have to export routes favourites and profiles and import again?

Clear data in Android app settings is like a clean install of the Android app.
So need first to export / backup your data and import / restore them later.

However if the device has storage problem and you have the data in another device,
it is better to restore from a "good" backup from the other device.

**Reply by DolomitesTom** [reply]
The problem is that the good backup is old. There are already a lot of changes in the new device. 

I don't believe that I have a storage problem.

**Reply by devemux86** [reply]
As a general rule, please always back up your data.

Don't trust anyone, mistakes can be made anywhere. :slightly_smiling_face:

**Comment by DolomitesTom** [comment]
My suspicion was that it gets confused after creating more a route groups with the resorting.
![Screenshot_20240506_095207_Total Commander](https://github.com/devemux86/cruiser/assets/121371629/ccc557a9-adbd-4194-b999-f67b483546f7)

**Reply by devemux86** [reply]
Did you change the data in the file system with a file manager?

This should not be done, just use Cruiser to manage the data.

**Reply by DolomitesTom** [reply]
No only as described created new folders and rearranged routes to these folders by using change group function. 

Also the properties didn't match anymore after this to the routes 

![Screenshot_20240506_095610_Cruiser](https://github.com/devemux86/cruiser/assets/121371629/aec5a65b-1e36-40bc-9d0f-4334c3788b7c)


For example I created a group called Mosel and not touched any route of the RdGA folder. Now I find them in the new group Mosel.

**Comment by DolomitesTom** [comment]
This is the message that comes up temporarily. 
The problem happens really everytime when changing group of route.
![Screenshot_20240506_100603_Cruiser](https://github.com/devemux86/cruiser/assets/121371629/8d432bb4-d1e7-4f6a-a1aa-e2c06f198c11)

**Comment by DolomitesTom** [comment]
After this rearranging the app crashes now when I try to load a route.

**Comment by devemux86** [comment]
@DolomitesTom thanks for the extra details.

Unfortunately I cannot reproduce it here.

Can you please send me your routes so I can make the same tests?
Go to the 1st tab with the groups, select "Export | All" and send me the zip.

Or if you are comfortable with the Android file manager,
can zip the folder "routes" and send me the zip.

**Comment by DolomitesTom** [comment]
After re-installation where do i find the POI download and activation menu?

**Reply by devemux86** [reply]
@DolomitesTom 

POI are automatically downloaded and activated with the Mapsforge maps.

If you need their download manager, enable the "Extended menu" in "Application | Advanced" settings.
Then look for "Offline POI" in the map menu.

**Reply by DolomitesTom** [reply, marked as answer]
thanks. I have no idea where this trouble came from. After deleting app cache and data and re-importing everything, I can't reproduce the error. Maybe Android took back some rights which prevented the app from loading and saving the changes correctly? But normally this only happens with apps not used often and I use Cruiser several times daily :-)

**Reply by devemux86** [reply]
@DolomitesTom something strange happened on your device.

You sent me the routes folder, but all routes (except 1) are empty with 0 size!
How this happened?

Anyway, I will add a check in the app to avoid processing "invalid" files.

**Reply by DolomitesTom** [reply]
OK, have no idea how this worked, am just happy that it works again.
